packages/base/src/commands.ts

```diff
@@ -1167,6 +1189,160 @@ export function addCommands(
     }
   });
 
+  commands.addCommand(CommandIDs.downloadGeoJSON, {
+    label: trans.__('Download as GeoJSON'),
+    isEnabled: () => {
+      const selectedLayer = getSelectedLayer(tracker);
+      return selectedLayer
+        ? ['VectorLayer', 'ShapefileLayer'].includes(
+            selectedLayer.selectedLayer.type
+          )
+        : false;
+    },
+    execute: async () => {
+      const selectedData = getSelectedLayer(tracker);
+      if (!selectedData) {
+        return;
+      }
+
+      const { model, selectedLayer, sources } = selectedData;
+      const exportSchema = {
+        ...(formSchemaRegistry.getSchemas().get('ExportGeoJSONSchema') as IDict)
+      };
+
+      const formValues = await new Promise<IDict>(resolve => {
+        const dialog = new ProcessingFormDialog({
```

Comment:
We may want to rename this class, since it's more general purpose than only processing:

```suggestion
        const dialog = new FormDialog({
```